### ISSUE TITLE
Better error handling for misformed tensors

### ISSUE THREAD

**Issue by yebai** [TuringLang/ThArrays.jl#7]
Running the following produces a seg fault, although the user isn't expected to create an Int tensor with gradient flag set to true. Maybe throw an error instead of producing a segfault? 

```julia
julia> x = Tensor(3, requires_grad=true)
libc++abi.dylib: terminating with uncaught exception of type c10::Error: Only Tensors of floating point dtype can require gradients (set_requires_grad at /Users/hg344/.julia/packages/ThArrays/67Vge/csrc/libtorch/include/torch/csrc/autograd/variable.h:211)


signal (6): Abort trap: 6
in expression starting at REPL[25]:1
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Allocations: 5627663 (Pool: 5626106; Big: 1557); GC: 12
zsh: abort      julia
```

**Comment by KDr2** [comment]
Fixed by https://github.com/TuringLang/ThArrays.jl/commit/36eb6d876abaa3ac9d2fa360588be56ea79a368e.

But must call `handle_error_in_julia()` after `using ThArrays`, failed to call it in the `__init__` method, I think it's a bug of `Libdl`.

**Comment by KDr2** [comment]
@JuliaRegistrator register

**Comment by KDr2** [comment]
@JuliaRegistrator register

**Comment by yebai** [comment]
> But must call handle_error_in_julia() after using ThArrays, failed to call it in the __init__ method, I think it's a bug of Libdl.

That's a bit strange since we're using `__init__` in `Turing.jl` as well and it works well.

**Comment by KDr2** [comment]
`__init__ ` is always being called while the module is imported, but is put a `ccall` in it (directly or indirectly), an error occurs, even after the statement which loads the shared library.

**Comment by yebai** [comment]
Can we call the function `handle_error_in_julia` outside `__init__`? This might result in `handle_error_in_julia` called every time when the module is imported, but I guess that is fine?

**Comment by KDr2** [comment]
Found a solution: call that function in `__init__` with `@async`.

**Comment by KDr2** [comment]
@JuliaRegistrator register
